Jenkinsfile

```
@@ -1,7 +1,7 @@
 pipeline {
 	options {
 		timeout(time: 45, unit: 'MINUTES')
-		buildDiscarder(logRotator(numToKeepStr:'10'))
+		buildDiscarder(logRotator(numToKeepStr:'10', artifactNumToKeepStr: '2'))
```

Maybe we should simply discard old build after e.g. 30 days anyways?

---

Does not hurt, but artifacts should still be discarded (and not wait for 30 days to lapse).

---

Yes but even 2 artifacts are wasted space for older branches no one really looks at... we mostly retain them for historic reasons or if one really wants to back port things so I think they can be pruned completely probably if nothing changes for a long time.

---

I never tested which option has priority. In a perfect world, artifacts of builds that are older than 30 days would be discarded, despite the option to keep 2 builds with artifacts. Realistically, limiting the number of builds saves more disk space for most use-cases, though.

Example:
* artifacts size: 100MB
* 10 jobs with 10 builds for 30 days => 10GB
* 10 jobs with max 2 builds with artifacts =>  2GB

This example applies to a standard CI build of a main branch where development happens for a long time. It's a different story for a branch that is only used temporarily, but is then kept around forever and never gets cleaned up. So cleaning up old branches is a best practice as well. :)

---

Does `artifactNumToKeepStr` in contrast to `numToKeepStr` mean that the artifacts for the builds 3 to 10 are discarded and only the other data of the run are kept?

I think for PR builds it's fine to even just keep the artifacts of the last build, but for the master it might be good to retain more. But I think it's sufficient to keep only 5 builds in general.

So my suggestion would be

```suggestion
		buildDiscarder(logRotator(numToKeepStr:'5', artifactNumToKeepStr: 'master'.equals(env.BRANCH_NAME) ? '5' : '1' ))
```

---

> Does artifactNumToKeepStr in contrast to numToKeepStr mean that the artifacts for the builds 3 to 10 are discarded and only the other data of the run are kept?

Yes.